### PR TITLE
[React] network selector revamp

### PR DESCRIPTION
**Pull request by MananTank**

### CODE REVIEW

File: packages/react-core/src/core/utils/updateChainRpcs.ts [hunk 1]
```diff
@@ -5,8 +5,12 @@ import { getChainRPCs, ChainRPCOptions, Chain } from "@thirdweb-dev/chains";
...
  */
 export const updateChainRPCs = (chain: Chain, options: ChainRPCOptions) => {
-  return {
-    ...chain,
-    rpc: getChainRPCs(chain, options),
-  };
+  try {
+    return {
+      ...chain,
+      rpc: getChainRPCs(chain, options),
```
Comment (iketw) [inline packages/react-core/src/core/utils/updateChainRpcs.ts:11]: Do you know why is this throwing?

Comment (MananTank) [reply packages/react-core/src/core/utils/updateChainRpcs.ts:11]: @iketw throws when passing a `Chain` with empty `rpc` array

Comment (MananTank) [reply packages/react-core/src/core/utils/updateChainRpcs.ts:11]: In dashboard, we are gonna pass all 700+ chains as supportedChains and some of them have empty rpc array

**Pull request merged**